src/components/PeriodDimension/PeriodSelector.js

```diff
@@ -0,0 +1,187 @@
+                            })
+                            this.setState({
+                                allPeriods: getRelativePeriodsOptionsById(
+                                    filter.value
+                                ).periods,
+                            })
+                        }}
+                    />
+                ) : (
+                    <FixedPeriodFilter
+                        currentPeriodType={this.state.fixedFilter.periodType}
+                        currentYear={this.state.fixedFilter.year}
+                        onSelectPeriodType={periodType => {
+                            this.onSelectFixedPeriods({
+                                periodType,
+                                year: this.state.fixedFilter.year,
+                            })
+                        }}
+                        onSelectYear={year => {
+                            this.onSelectFixedPeriods({
+                                periodType: this.state.fixedFilter.periodType,
+                                year,
+                            })
+                        }}
+                    />
+                )}
+            </div>
+            <style jsx>{styles}</style>
+        </>
+    )
+
+    onSelectFixedPeriods = fixedFilter => {
+        this.setState({
+            fixedFilter,
+            allPeriods: getFixedPeriodsOptionsById(
+                fixedFilter.periodType.value
+            ).generator.generatePeriods(
+                fixedPeriodConfig(Number(fixedFilter.year))
+            ),
+        })
+    }
+
+    renderEmptySelection = () => (
+        <>
+            <p className="emptySelection">{i18n.t('No periods selected')}</p>
+            <style jsx>{styles}</style>
+        </>
+    )
+
+    render = () => (
+        <Transfer
+            onChange={({ selected }) => {
+                this.setState({ selectedPeriods: selected })
+                this.props.onSelect(selected)
+            }}
+            selected={this.state.selectedPeriods}
+            leftHeader={this.renderHeader()}
+            enableOrderChange
+            height="512px"
+            optionsWidth="420px"
+            selectedWidth="298px"
+            selectedEmptyComponent={this.renderEmptySelection()}
+            rightFooter={this.props.rightFooter}
+            // TODO: Add rightHeader "Selected Periods" once the Transfer component supports this (https://github.com/dhis2/ui-core/issues/885)
+        >
+            {this.state.allPeriods.map(item => (
+                <TransferOption
+                    label={item.getName()}
+                    value={item.id}
+                    key={item.id}
+                    icon={PeriodIcon}
+                />
+            ))}
+        </Transfer>
+    )
+}
+
+PeriodSelector.propTypes = {
+    onSelect: PropTypes.func.isRequired,
+    rightFooter: PropTypes.node,
+    selectedItems: PropTypes.arrayOf(PropTypes.object),
```

Comment:
One thing that always confused me when I'd been away from this code and then came back to it again, was having both `selectedItems` and `selectedPeriods`.  I thought about renaming `selectedItems` to `initialSelectedPeriods`, to clarify its purpose. What do you think?